save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/ProjectStatisticMenu.kt

```diff
@@ -0,0 +1,155 @@
+@file:Suppress("FILE_NAME_MATCH_CLASS", "FILE_WILDCARD_IMPORTS", "LargeClass")
+
+package org.cqfn.save.frontend.components.basic
+
+import org.cqfn.save.agent.TestSuiteExecutionStatisticDto
+import org.cqfn.save.domain.TestResultStatus
+import org.cqfn.save.frontend.components.tables.tableComponent
+import org.cqfn.save.frontend.externals.chart.DataPieChart
+import org.cqfn.save.frontend.externals.chart.pieChart
+import org.cqfn.save.frontend.externals.chart.randomColor
+import org.cqfn.save.frontend.utils.apiUrl
+import org.cqfn.save.frontend.utils.decodeFromJsonString
+import org.cqfn.save.frontend.utils.get
+import org.cqfn.save.frontend.utils.unsafeMap
+
+import org.w3c.fetch.Headers
+import react.*
+import react.dom.div
+import react.dom.td
+import react.table.columns
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+
+/**
+ * ProjectStatisticMenu component props
+ */
+external interface ProjectStatisticMenuProps : Props {
+    /**
+     * Id of execution
+     */
+    var executionId: Long?
+
+    /**
+     * list of tests
+     */
+    var latestExecutionStatisticDtos: List<TestSuiteExecutionStatisticDto>?
+
+    /**
+     * Flag to open Menu
+     */
+    var isOpen: Boolean?
+}
+
+/**
+ * @param onOpenMenuStatistic
+ * @return ReactElement
+ */
+@Suppress("TOO_LONG_FUNCTION", "LongMethod", "MAGIC_NUMBER")
+fun projectStatisticMenu(
+    onOpenMenuStatistic: (isOpen: Boolean) -> Unit,
+) =
+        fc<ProjectStatisticMenuProps> { props ->
+
+            val (latestExecutionStatisticDtos, setLatestExecutionStatisticDtos) = useState(props.latestExecutionStatisticDtos)
+            val scope = CoroutineScope(Dispatchers.Default)
+
+            if (props.isOpen == true) {
+                useEffect(listOf<dynamic>(props.executionId, props.latestExecutionStatisticDtos)) {
+                    scope.launch {
+                        val testLatestExecutions = get(
+                            url = "$apiUrl/testLatestExecutions?executionId=${props.executionId}&status=${TestResultStatus.PASSED}",
+                            headers = Headers().also {
+                                it.set("Accept", "application/json")
+                            },
+                        )
+                            .unsafeMap {
+                                it.decodeFromJsonString<List<TestSuiteExecutionStatisticDto>>()
+                            }
+                        setLatestExecutionStatisticDtos(testLatestExecutions)
+                    }
+                    onOpenMenuStatistic(false)
```

Comment:
Are you using this flag as a workaround to avoid request on rerender? Method name is very confusing.